Lib/logging/config.py

```diff
@@ -674,6 +680,7 @@ def configure_formatter(self, config):
                 config['()'] = factory
                 result = self.configure_custom(config)
         else:
+            self._check_config_keys(config, ('format', 'datefmt', 'style', 'validate', 'class'))
```

Comment:
Since `key in tuple()` is `O(n)` and `key in set()` is `O(1)`, maybe we can change the tuple to a set, to obtain a potential performance benefit.

```suggestion
            self._check_config_keys(config, {'format', 'datefmt', 'style', 'validate', 'class'})
```

Reply:
Actually, for smaller sets of data, `set` often performs worse than `tuple`. You can run a quick benchmark to check it out. There's a cost for the O(1) performance - a larger constant. Converting this to set won't gain us any performance.